File: ios/TeamHome/TeamHome/Models/Team.swift
//
//  Team.swift
//  TeamHome
//
//  Created by Daniela Parra on 1/10/19.
//  Copyright © 2019 Lambda School under the MIT license. All rights reserved.
//

import Foundation

class Team {
    let identifier: UUID
    let name: String
    let admins: [User] //only users where admin = true
    let premium: Bool
    
    init(name: String, admins: [User], premium: Bool, identifier: UUID) {
        self.identifier = identifier
        self.name = name
        self.admins = admins
        self.premium = premium
    }
    
}

File: backend/models/Team.js
const { Schema, model } = require('mongoose');

const Team = new Schema({
	name: { type: String, trim: true, required: true },
	users: [
		{
			user: { type: Schema.Types.ObjectId, ref: 'User', required: true },
			admin: { type: Boolean, default: false }
		}
	],
	premium: { type: Boolean, default: false }
});

module.exports = model('Team', Team);
